### DISCUSSION TITLE
🤖 AI access for the hackathon

### DISCUSSION THREAD

**Discussion by aaronpowell** [microsoft/hack-together-dotnet#23] [Announcements]
I hope everyone is excited to get going with the hackathon!

We know many of you are wanting to use AI for your projects and to facilitate this we're offering no-cost access to [Azure OpenAI Service](https://learn.microsoft.com/azure/ai-services/openai/overview) for the duration of the hackathon - you don't even need an Azure account.

To get access [fill out this form](https://aka.ms/hacktogether/dotnet/AzureOpenAIAccess) to apply for access, then you will receive an email from `DoNotReply@hacktogether.net` within 1 business day.

Please be responsible with the access you are granted, as we will revoke keys that that are being abused.

Happy hacking!
- The Great .NET 8 Hackathon team

**Comment by aaronpowell** [comment]
## Using the proxy

Using the proxy works the same as when the Azure OpenAI Service, so you can refer to the samples [on the Azure OpenAI Service docs](https://learn.microsoft.com/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line%2Cpython&pivots=programming-language-csharp).

The connection details are as follows:

- **Endpoint**: https://aoai.hacktogether.net/v1/api
- **API Key**: `guid/github-username`
  - This is emailed to you when you apply via the form above

You can also access a playground for the proxy at https://aoai.hacktogether.net

**Reply by shreenivaska** [reply]
Thanks for the reply. In lesson 4, it asks to give API key. Which api key it refers here (prompt engineering fundamentals chapter).

**Reply by codemillmatt** [reply]
Hi @shreenivaska - assuming you're using the .NET SDK, you would enter the key as "{KEY YOU WERE SENT IN THE EMAIL}/{YOUR GH USERNAME}".  You can see a quick example of using the .NET SDK here: https://aka.ms/hacktogether/dotnet/proxy-instructions

**Reply by shreenivaska** [reply]
It is python code that I got forked. It does not have any proxy or endpoint.

**Reply by codemillmatt** [reply]
@shreenivaska - for this hacktogether project it would be best to use .NET. (Python should work, though I haven't tried it - but this is all about .NET here 😉 ). Check this out for instructions including a code sample: https://github.com/microsoft/hack-together-dotnet/blob/main/OPENAI-README.md

**Reply by codemillmatt** [reply]
Looks like we might be dealing with a transient issue @shreenivaska - I just tried it and it worked for me.

Fwiw - we're limited to 2048 tokens per session.

**Reply by shreenivaska** [reply]
It worked. I just used AzureOpenAI and proxy URL.

**Reply by tinycrops** [reply]
is there a way to hack the workshop playground? is there a repo available for it?

**Comment by abimr** [comment]
Azure OpenAI Services doesn't expose the new OpenAI Assistants API yet.

Are we allowed to directly use the OpenAI API instead?

**Comment by aaronpowell** [comment]
No, assistants aren't available on Azure OpenAI Service.

Yes, you can use OpenAI directly if you'd like.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by edilma** [comment]
Hello there:
My api key doesn't work.   I get the following error.  What can i do?   ---> Azure.RequestFailedException: Incorrect API key provided: xxxxx*******************************yyy. 
Status: 401 (Unauthorized)
ErrorCode: invalid_api_key

Can i request a new key to be generated or activate mine?  How do i do that?  Thanks

**Reply by edilma** [reply]
I did follow the instructions

**Reply by edilma** [reply]
I tried in the playground and it works there.  I guess is my code.  Thanks

**Reply by codemillmatt** [reply]
Hi @edilma - feel free to create a gist with your code - or a link to your repo and I'd be happy to take a peek at it and let you know if anything jumps out at me. Fwiw - we have been seeing transient errors when load has been high. But glad to hear your key is working.
